app/src/play/java/de/danoeh/antennapod/dialog/CustomMRControllerDialog.java

```diff
@@ -124,38 +165,89 @@ public View onCreateMediaControlView(Bundle savedInstanceState) {
-                        desiredHeight = MeasureSpec.makeMeasureSpec((int) (intrHeight * scale + 0.5f), MeasureSpec.EXACTLY);
                     }
+                    super.onMeasure(desiredWidth, heightMeasureSpec);
                 }
+            };
+            artParams = new LinearLayout.LayoutParams(
+                    ViewGroup.LayoutParams.WRAP_CONTENT,
+                    ViewGroup.LayoutParams.MATCH_PARENT);
+            MarginLayoutParamsCompat.setMarginStart(artParams,
+                    getContext().getResources().getDimensionPixelSize(R.dimen.media_router_controller_playback_control_horizontal_spacing));
+        } else {
+            artView = new ImageView(getContext()) {
+                @Override
+                protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
+                    int desiredHeight = heightMeasureSpec;
+                    if (MeasureSpec.getMode(heightMeasureSpec) != MeasureSpec.EXACTLY) {
+                        Drawable drawable = getDrawable();
+                        if (drawable != null) {
+                            int originalWidth = MeasureSpec.getSize(widthMeasureSpec);
+                            int intrHeight = drawable.getIntrinsicHeight();
+                            int intrWidth = drawable.getIntrinsicWidth();
+                            float scale;
+                            if (intrHeight*16 > intrWidth*9) {
+                                // image is taller than 16:9
+                                scale = (float) originalWidth * 9 / 16 / intrHeight;
+                            } else {
+                                // image is more horizontal than 16:9
+                                scale = (float) originalWidth / intrWidth;
+                            }
+                            desiredHeight = MeasureSpec.makeMeasureSpec(
+                                    Math.round(intrHeight * scale),
+                                    MeasureSpec.EXACTLY);
+                        }
+                    }
+                    super.onMeasure(widthMeasureSpec, desiredHeight);
+                }
+            };
+            artParams = new LinearLayout.LayoutParams(
+                    ViewGroup.LayoutParams.MATCH_PARENT,
+                    ViewGroup.LayoutParams.WRAP_CONTENT);
+        }
+        artView.setTag(landscape);
 
-                super.onMeasure(widthMeasureSpec, desiredHeight);
-            }
-        };
-        artView.setLayoutParams(new FrameLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT,
-                ViewGroup.LayoutParams.WRAP_CONTENT));
         artView.setScaleType(ImageView.ScaleType.FIT_CENTER);
         artView.setOnClickListener(onClickListener);
 
+        artView.setLayoutParams(artParams);
         rootView.addView(artView);
-        View playbackControlLayout = View.inflate(getContext(), R.layout.media_router_controller, rootView);
+
+        ViewGroup wrapper = rootView;
+
+        if (landscape) {
```

Comment:
did you come up with this yourself? is this a common solution to problem X? any interwebs resources to read up on this?